server/background_tasks.go

```diff
@@ -338,3 +353,99 @@ func (s *Server) StartBackgroundTasks() error {
 	}
 	return nil
 }
+
+func (s *Server) UploadMsgsToTierStorage() {
+	ticker := time.NewTicker(time.Duration(configuration.TIERED_STORAGE_TIME_FRAME_SEC) * time.Second)
+	for range ticker.C {
+		isSuccessUploadToS3 := false
+		lock.Lock()
+		if len(tierStorageMsgsMap.m) > 0 {
+			err := UploadToTier2Storage()
+			if err != nil {
+				isSuccessUploadToS3 = false
+				serv.Errorf("UploadMsgsToTierStorage: " + err.Error())
+			} else {
+				isSuccessUploadToS3 = true
+			}
+		}
+
+		if isSuccessUploadToS3 {
+			for i, msgs := range tierStorageMsgsMap.m {
+				for _, msg := range msgs {
+					reply := msg.ReplySubject
+					s.sendInternalAccountMsg(s.GlobalAccount(), reply, []byte(_EMPTY_))
+				}
+				tierStorageMsgsMap.Delete(i)
+			}
+		}
+		lock.Unlock()
+	}
+}
+
+func (s *Server) ApiRequestToJetstreamFetchTieredStorageMsgs() {
+	ticker := time.NewTicker(1 * time.Second)
+	for range ticker.C {
+		if isTierStorageConsumerCreated && isTierStorageStreamCreated {
+			durableName := TIERED_STORAGE_CONSUMER
+			subject := fmt.Sprintf(JSApiRequestNextT, tieredStorageStream, durableName)
+			reply := durableName + "_reply"
+			amount := 1000
+			req := []byte(strconv.FormatUint(uint64(amount), 10))
+			serv.sendInternalAccountMsgWithReply(serv.GlobalAccount(), subject, reply, nil, req, true)
+		}
+	}
+}
+
+func (s *Server) ListenForTieredStorageMessages() error {
+	tierStorageMsgsMap = NewConcurrentMap[[]StoredMsg]()
+
+	reply := TIERED_STORAGE_CONSUMER + "_reply"
+	err := serv.queueSubscribe(reply, reply+"_sid", func(_ *client, subject, reply string, msg []byte) {
+		go func(subject, reply string, msg []byte) {
+			//This if ignores case: 409 Exceeded MaxWaiting
+			if reply != "" {
+				rawMsg := strings.Split(string(msg), CR_LF+CR_LF)
+				var tieredStorageMsg TieredStorageMsg
+				err := json.Unmarshal([]byte(rawMsg[1]), &tieredStorageMsg)
```

Comment:
you are accessing [1] index and it is not necessarily exist so please check the length of the splitted raw message